Marlin/src/inc/SanityCheck.h

```diff
@@ -1419,6 +1419,19 @@ static_assert(COUNT(arm) == LOGICAL_AXES, "AXIS_RELATIVE_MODES must contain " _L
     static_assert(PROBING_MARGIN_RIGHT >= 0, "PROBING_MARGIN_RIGHT must be >= 0.");
   #endif
 
+  #define static_assert_within(varname, x, min, max) static_assert(WITHIN(x, min, max), #varname " must be within " #min " and " #max)
```

Comment:
I wish there was a `format(...)` function so the assert could report actual values for both `x`, `min` and `max`

Reply:
Simplified assert here is fine as long as it is clear. e.g., Just emit a message the the offset is too big (larger than the bed size).